kong/conf_loader/init.lua

```diff
@@ -11,11 +11,55 @@ local tablex = require "pl.tablex"
 local utils = require "kong.tools.utils"
 local log = require "kong.cmd.utils.log"
 local env = require "kong.cmd.utils.env"
-local ip = require "resty.mediador.ip"
+local ipmatcher = require "resty.ipmatcher"
 
 
+local string = string
 local fmt = string.format
+local type = type
 local concat = table.concat
+local tonumber = tonumber
+
+
+local function is_valid_ip_or_cidr(ip_or_cidr)
+  if type(ip_or_cidr) ~= "string" then
+    return false
+  end
+
+  if ipmatcher.parse_ipv4(ip_or_cidr)
+  or ipmatcher.parse_ipv6(ip_or_cidr)
+  then
+    return true
+  end
+
+  local p = string.find(ip_or_cidr, "/", 1, true)
+  if not p then
+    return false
+  end
+
+  local block = tonumber(string.sub(ip_or_cidr, p + 1), 10)
+  if not block then
+    return false
+  end
+
+  local ip = string.sub(ip_or_cidr, 1, p - 1)
+  if ipmatcher.parse_ipv4(ip) then
+    for i = 0, 32 do
+      if block == i then
+        return true
+      end
+    end
+
+  elseif ipmatcher.parse_ipv6(ip) then
+    for i = 0, 128 do
+      if block == i then
+        return true
+      end
+    end
+  end
+
+  return false
+end
```

Comment:
Noob question: If `ipmatcher` could not parse, shouldn't we return `false` and that's it? What is the purpose of all the extra logic here? We didn't have this logic here before. I assume it's because `ipmatcher` can't do what mediador was doing before. It seems like we are implementing something new. Could it be a patch (bug or feature) to the `ipmatcher` lib?

Reply:
Yes, so `ipmatched.parse_ipv4/6` only takes ip addresses. At the very top we check if it was ip address, if it was not, perhaps it is a cidr block, thus we split the input string in two `/` and then parse if the address is ipv4, if it is ipv4 we also check that `/xx` is between 0-32 (all possible masks). For ipv6 there are other notations too, but atm we just support `/xxx` notation where the xxx is between 0-128.

So logic of this is:
1. is it ipv4 or ipv6
2. is it ipv4 or ipv6 cidr block

Reply:
Yes, I wish the library itself could be better in this. There is `ipmatcher.new` but on my testing it was not good.

Reply:
ahh I see, now I remember about this: https://github.com/api7/lua-resty-ipmatcher/issues/11